### PR TITLE
Clarify wording of Connect's Telemetry FAQ

### PR DESCRIPTION
**Pull request by timothyb89**
This clarifies the wording of Connect's Telemetry FAQ which implied that it's impossible in all cases to deanonymize event fields, which may not be true if the cluster's internal datastore is compromised.

This attempts to paint a more accurate picture of the deanonymization threat model, since ultimately the cluster's random ID can be used in a dictionary-style attack if known or leaked.

### CODE REVIEW

File: docs/pages/includes/teleport-connect-telemetry.mdx [hunk 1]
```diff
@@ -16,6 +16,6 @@ On login, we also collect some device-related data:
 Additionally, we ask for a job role (answer is optional).
 
 The full list of events and collected data is defined as [protocol buffer messages in the Teleport source](https://github.com/gravitational/teleport/tree/branch/v(=teleport.major_version=)/proto/prehog/v1alpha/connect.proto).
-We do not track the details of those events but merely that the given event took place. Each event includes
-the cluster name and user name anonymized with HMAC with the internal cluster ID as the key – it’s impossible for us
-to resolve that anonymized token back to a particular cluster or user.
+We do not track the details of those events but merely that the given event took place. Each event includes the cluster
+name and user name anonymized with HMAC using the cluster's internal random UUID as the key. It is infeasible to
```
Comment (strideynet) [inline docs/pages/includes/teleport-connect-telemetry.mdx:20]: It might be worth us explaining why we want to track the anonymised cluster and user ID.

**Review by strideynet — APPROVED** [review]
This copy is definitely more accurate to what it actually occurring. I do wonder if there's any value in us explaining why we need these values though.

File: docs/pages/includes/teleport-connect-telemetry.mdx (hunk 1 above)
Comment (gzdunek) [reply docs/pages/includes/teleport-connect-telemetry.mdx:20]: It allows us to combine the Connect events and Teleport events (although we haven't done it yet), but I'm not sure if this is worth mentioning.

**Review by gzdunek — APPROVED** [review]

File: docs/pages/includes/teleport-connect-telemetry.mdx (hunk 1 above)
Comment (strideynet) [reply docs/pages/includes/teleport-connect-telemetry.mdx:20]: It's somewhat my personal opinion that if we want people to opt-in to telemetry, this works better when we provide individual justification for data-points collected - it demonstrates that we are trying to collect the least data possible whilst still implementing useful Telemetry and that we are being responsible about telemetry. For the Machine ID telemetry I explained what data is collected that allows the linking of events, and what data is explicitly not collected: https://goteleport.com/docs/machine-id/reference/telemetry/

**Review by ptgott — APPROVED** [review]

File: docs/pages/includes/teleport-connect-telemetry.mdx (hunk 1 above)
Comment (timothyb89) [reply docs/pages/includes/teleport-connect-telemetry.mdx:20]: I'm inclined to leave this as-is for now, but maybe we can consider making a unified docs page for our telemetry data that explains everything in detail? I think I'd rather put the wordsmithing effort into that than an FAQ entry that technically only covers Connect.

Comment (gzdunek) [reply docs/pages/includes/teleport-connect-telemetry.mdx:20]: Hmm having everything in one place sounds nice.

@timothyb89 could you backport this PR to v12? I forgot about it earlier.

**Comment by alexfornuto** [comment]
Does  this only relate to the next release, or should we backport it so it goes to the docs site before then?

File: docs/pages/includes/teleport-connect-telemetry.mdx (hunk 1 above)
Comment (timothyb89) [reply docs/pages/includes/teleport-connect-telemetry.mdx:20]: Yup, I'll backport it

**Comment by timothyb89** [comment]
> Does this only relate to the next release, or should we backport it so it goes to the docs site before then?

We have some Telemetry in v12, so it should be backported. I'll do so shortly.

**Pull request merged**